src/mousev1_to_nwb/abdeladim_2023/abdeladim_2023imaginginterface.py

```diff
@@ -0,0 +1,128 @@
+            # dateparse can't cope with this representation, so using strptime directly
+            extracted_session_start_time = datetime.datetime.strptime(
+                self.image_metadata["epoch"], "[%Y %m %d %H %M %S.%f]"
+            )
+            metadata["NWBFile"].update(session_start_time=extracted_session_start_time)
+
+        # Extract many scan image properties and attach them as dic in the description
+        ophys_metadata = metadata["Ophys"]
+        two_photon_series_metadata = ophys_metadata["TwoPhotonSeries"][device_number]
+        if self.image_metadata is not None:
+            extracted_description = json.dumps(self.image_metadata)
+            two_photon_series_metadata.update(description=extracted_description)
+        channel_name_without_space = self.channel_name.replace(" ", "")
+        # indicators = {"Channel1": "GCaMP6f", "Channel2": "tdTomato"} # Not specified
+
+        channel_metadata = {
+            "Channel1": {
+                "name": "Green",
+                #"emission_lambda": 513.0, # Not specified
+                "description": "Green channel of the microscope",
+            },
+            "Channel2": {
+                "name": "Red",
+                #"emission_lambda": 581.0,# Not specified
+                "description": "Red channel of the microscope",
+            },
+        }[channel_name_without_space]
+
+        optical_channel_metadata = channel_metadata
+
+        device_name = "CustomMicroscope"
+        metadata["Ophys"]["Device"][0].update(
+            name=device_name,
+            description="The mesoscale read/write platform was custom-built around a 2P random-access fluorescence mesoscope previously described in detail (Sofroniew 2016)",
+            manufacturer="Thorlabs Inc.", 
+        )
+
+        # indicator = indicators[channel_name_without_space]
+        imaging_plane_name = f"ImagingPlane{channel_name_without_space}Plane{self.plane_name}"
+        imaging_plane_metadata = metadata["Ophys"]["ImagingPlane"][0]
+        imaging_plane_metadata.update(
+            name=imaging_plane_name,
+            optical_channel=[optical_channel_metadata],
+            device=device_name,
+            excitation_lambda=920.0,
+            #indicator=indicator,
+            imaging_rate=self.imaging_extractor.get_sampling_frequency(),
+        )
+
+        two_photon_series_metadata = metadata["Ophys"]["TwoPhotonSeries"][0]
+        two_photon_series_metadata.update(
+            name=f"TwoPhotonSeries{channel_name_without_space}Plane{self.plane_name}",
+            imaging_plane=imaging_plane_name,
+            scan_line_rate=1 / float(self.image_metadata["SI.hRoiManager.linePeriod"]),
+            rate=self.imaging_extractor.get_sampling_frequency(),
+            description=f"Two photon series acquired with {self.channel_name} at plane {self.plane_name}",
+            unit="n.a.",
+            dimension=self.imaging_extractor.get_image_size(),
+        )
```

Comment:
I think dimension is automatically added from NeuroConv in `get_nwb_imaging_metadata` also unit is "n.a." by default, no need to specify it here as well.